### PR TITLE
lammps and python312Packages.lammps related fixes

### PR DESCRIPTION
**Pull request by doronbehar**
Found out this was needed when trying to override the python package to use a LAMMPS compiled with GPU support.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/la/lammps-mpi/package.nix [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (markuskowa) [inline pkgs/by-name/la/lammps-mpi/package.nix:1]: The definition of `lamps-mpi` should probably still go into `all-packages.nix` (if I read this correctly: https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#readme)

Comment (doronbehar) [reply pkgs/by-name/la/lammps-mpi/package.nix:1]: https://github.com/NixOS/nixpkgs/pull/342621

Comment (doronbehar) [reply pkgs/by-name/la/lammps-mpi/package.nix]: @markuskowa I moved the `lammps-mpi` move away from `all-packages.nix` to https://github.com/NixOS/nixpkgs/pull/342621 . As I want to get the important commits of this PR merged, as it seems we agree upon them. I rebased, and removed that commit.

**Review by markuskowa — APPROVED** [review]
LGTM.
Sorry for the delay. The pkgs/by-name infrastructure is not my area of expertise. I'll leave that to @infinisil to make architectural decisions.

**Comment by doronbehar** [comment]
> LGTM.
> Sorry for the delay. The pkgs/by-name infrastructure is not my area of expertise.

It's OK, no need to be sorry :). This was an open debate that legitimately held this PR open. Now we can discuss it elsewhere (in https://github.com/NixOS/nixpkgs/pull/342621)

**Pull request merged**